Src/tim.c

```diff
@@ -52,12 +52,12 @@ void MX_TIM16_Init(void)
   TIM_OC_InitTypeDef sConfigOC;
   TIM_BreakDeadTimeConfigTypeDef sBreakDeadTimeConfig;
 
-  htim16.Instance = TIM16;
-  htim16.Init.Prescaler = 0;
+  htim16.Instance = TIM16; //setting the compare register
+  htim16.Init.Prescaler = 1024; //brings freq down to 46875, need it at 50. So Compare match register 936.5 (+1) to get interrupt freq = 50Hz
   htim16.Init.CounterMode = TIM_COUNTERMODE_UP;
-  htim16.Init.Period = 0;
+  htim16.Init.Period = 20; //20ms for 50Hz Freq
```

Comment:
![image](https://user-images.githubusercontent.com/32375512/106566960-0a77e600-64ff-11eb-9663-49479414aa18.png)

Take a look at this snippet from the datasheet (link: https://www.st.com/resource/en/reference_manual/dm00091010-stm32f030x4x6x8xc-and-stm32f070x6xb-advanced-armbased-32bit-mcus-stmicroelectronics.pdf)

Reply:
I implemented the ARR value for period as the Period is the number of steps and the ARR is the max value before reset. Not the most confident that I got that right.